hash/hash.go

```diff
@@ -0,0 +1,67 @@
+package hash
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"hash"
+
+	"github.com/mr-tron/base58"
+)
+
+// DefaultHash is a default hashing algorithm - "sha256".
+var DefaultHash = sha256.New
+
+// size is a default size for hashing algorithm.
+var size = DefaultHash().Size()
+
+// Digest represents the partial evaluation of a checksum.
+type Digest struct {
+	hash.Hash
+}
+
+// Sum appends the current checksum to b and returns the Hash.
+func (d *Digest) Sum(b []byte) Hash {
+	return Hash(d.Hash.Sum(b))
+}
+
+// A Hash is a type for representing hash with base58 encode and decode functions.
+type Hash []byte
+
+// New returns new hash from a given integer.
+func New() *Digest {
+	return &Digest{
+		Hash: DefaultHash(),
+	}
+}
+
+// Int returns a new hash from a given integer.
+func Int(h int) Hash {
```

Comment:
is it useful to have this function?
is it only for test?

Reply:
Only for tests